tests/base/test_prompts.py

```diff
@@ -132,3 +132,21 @@ class MyPrompt(BasePrompt):
 
     with pytest.raises(ValueError):
         MyPrompt(bad=1).messages()
+
+
+def test_base_prompt_list_attribute() -> None:
+    """Tests that attributes of type list are properly injected."""
+
+    class MyPrompt(BasePrompt):
+        prompt_template = """
+        {my_list}
+        {my_list_of_lists}
+        """
+
+        my_list: list[str]
+        my_list_of_lists: list[list[str]]
+
+    my_list = ["my", "list"]
+    my_list_of_lists = [my_list]
+    prompt = MyPrompt(my_list=my_list, my_list_of_lists=my_list_of_lists)
+    assert str(prompt) == "my\nlist\n['my', 'list']"
```

Comment:
hmm so we don't flatten lists of lists? With RAG, I have list[list[str]], depending on how many queries a user makes for similarity. that entire context will be put in the prompt.

Reply:
I guess I could flatten further, and then separate with "\n\n" but when does it end? Do we support `list[list[list[str]]]`?

Reply:
I figured for something like a nested list, if you want more than just `str(inner_list)` that you would want more control than the default anyway? e.g.

```python
@property
def formatted_list_of_lists(self) -> list[str]:
    return [
        f"Index: {i}\n{"\n".join([item for item in l])}\n"
        for i, l in enumerate(self.my_list_of_lists)
    ]
```

but I guess it would still be convenient for us to at least have some default handling of `list[list[...]]`?

Reply:
i think we can stop at `list[list[...]]` because thats a real use case.

Reply:
Done.